src/blockMonitor/blockProcessor.ts

```diff
@@ -104,26 +129,11 @@ export class BlockProcessor<TBlock extends IBlockStub> extends StartStopService
         return this.mBlockCache;
     }
 
-    /**
-     * Event emitted when a new block is mined and has been added to the BlockCache.
-     * It is not guaranteed that no block is skipped, especially in case of reorgs.
-     * Emits the block stub of the new head, and the previous emitted block in the ancestry of this block..
-     */
-    public static readonly NEW_HEAD_EVENT = "new_head";
-
-    /**
-     * Event that is emitted for each new blocks. It is emitted (in order from the lowest-height block) for each block
-     * in the ancestry of the current blockchain head that is deeper than the last block in the ancestry that was emitted
-     * in a previous NEW_HEAD_EVENT. The NEW_HEAD_EVENT for the latest head block is guaranteed to be emitted after all the
-     * NEW_BLOCK_EVENTs in the ancestry have been emitted.
-     * A NEW_BLOCK_EVENT might happen to be emitted multiple times for the same block in case of blokchain re-orgs.
-     */
-    public static readonly NEW_BLOCK_EVENT = "new_block";
-
     constructor(
         private provider: ethers.providers.BaseProvider,
         blockFactory: BlockFactory<TBlock>,
-        blockCache: BlockCache<TBlock>
+        blockCache: BlockCache<TBlock>,
+        private readonly store: BlockProcessorStore
     ) {
         super("block-processor");
 
```

Comment:
processBlockNumber is still the listener for the provider's "block" event, so no change there.

But even for callbacks to our internal events, since they will be called as
   `   listener(...);`
somewhere, the `this` inside the listener would be wrong without that (unless they are declared with arrow notation, of course).
Essentially the bind should be done for any callback (except perhaps small pure functions that don't use `this`, but they would probably be arrow functions anyway).